spyder/plugins/editor/widgets/completion.py

```diff
@@ -300,7 +300,8 @@ def keyPressEvent(self, event):
         alt = event.modifiers() & Qt.AltModifier
         shift = event.modifiers() & Qt.ShiftModifier
         ctrl = event.modifiers() & Qt.ControlModifier
-        modifier = shift or ctrl or alt
+        altgr = (key == 0)
```

Comment:
Hi @MaxGyver83 maybe instead of the `0` we could use here `Qt.Key_AltGr`? Also, maybe it could be worthy to add a comment above this line referencing the issue this change fixes, it could be something like:

```python
# Needed to properly handle Neo2 and other keyboard layouts
# See spyder-ide/spyder#11293
```

Reply:
Thanks for your comment!

The value of `key` equals `Qt.Key_AltGr` (=16781571) when the normal QWERTZ/QWERTY layout is active. So yes, we should also check for this value as it is a modifier key.

But this doesn't work for Neo2 (and others) because in these layouts _AltGr_ is not defined as **ISO_Level3_Shift** but as **ISO_Level5_Shift** (in Neo2 terminology that's level 4). Unfortunately, Qt seems not to have a value for ISO_Level5_Shift, so it sets `key` to 0.

I suppose there is no other key with the value of 0 that is supposed to close the autocompletion menu. So I'm going to make a new commit in order to check for `key` values 0 and `Qt.Key_AltGr` (=16781571). And I'm going to add a comment.